python/cuspatial/cuspatial/geometry/geoseries.py

```diff
@@ -0,0 +1,599 @@
+# Copyright (c) 2020-2021, NVIDIA CORPORATION
+
+import numbers
+from geopandas.geoseries import GeoSeries as gpGeoSeries
+import pandas as pd
+import numpy as np
+
+from shapely.geometry import (
+    Point,
+    MultiPoint,
+    LineString,
+    MultiLineString,
+    Polygon,
+    MultiPolygon,
+)
+
+import cudf
+from cudf.core.column import ColumnBase
+
+from cuspatial.io.geoseries_reader import GeoSeriesReader
+
+
+class GeoSeries(ColumnBase):
+    def __init__(self, data, name=None, index=None):
```

Comment:
Should this match the signature of `geopandas.GeoSeries`? https://geopandas.org/reference/geopandas.GeoSeries.html

Reply:
The only argument that it accepts is either a `geopandas.GeoSeries` or a `cuspatial.GeoSeries`, so, not at this time. It doesn't accept abstract data - that's GeoPandas's job.
 
 I can see an easy way to make its signature match GeoPandas, simply taking in whatever the `data` arg is, letting GeoPandas read it, then reading that with `cuspatial`, but the result would be unexpectedly slow - that is, not handled by GPU at all.
 
 We also don't support CRS yet. Down the road we definitely will. Handling abstract data in the above response, purely on GPU, is much less likely.